imod/mf6/package.py

```diff
@@ -533,17 +533,37 @@ def mask(self, domain: GridDataArray) -> Any:
         masked: Package
             The package with part masked.
         """
+
```

Comment:
This method is becoming really big. With the nested if and for loops it becomes hard to read.
You can try to split it up
Below is some untested code which probably needs a little bit of finetuning to get it to work:

```
def mask(self, domain: GridDataArray) -> Any:
   ....
    def can_be_masked(var, da):
        return self.skip_masking_dataarray(var) or len(da.dims) == 0 or set(da.coords).issubset(["layer"])
    
    maskable_vars = [var for var in self.dataset.data_vars.keys() if can_be_masked(var, self.dataset[var])]
    non_maskable_vars = [var for var in self.dataset.data_vars.keys() if not can_be_masked(var, self.dataset[var])]
    
    for var in non_maskable_vars:
        masked[var] = self.dataset[var]
    
    spatial_maskable_vars = [var for var in maskable_vars if da.coords in (spatial_dims)]
    non_spatial_maskable_vars = [var for var in maskable_vars if da.coords not in (spatial_dims)]
    
    for var in spatial_maskable_vars:
        masked[var] = mask_spatial_var(var)
    
    for var in non_spatial_maskable_vars:
        masked[var] = mask_non_spatial_var(var)



def mask_spatial_var(var):
    if len(da.dims) < len(da.coords):
        if "layer" in da.coords and "layer" not in da.dims:
            array_domain = domain.sel(layer=da.coords["layer"])
        if "layer" not in da.coords and "layer" in array_domain.coords:
            array_domain = domain.isel(layer=0)

    if issubclass(da.dtype.type, numbers.Integral):
        if var == "idomain":
            return da.where(array_domain > 0, other=array_domain)
        else:
            return da.where(array_domain > 0, other=0)
    elif issubclass(da.dtype.type, numbers.Real):
        return da.where(array_domain > 0)
    else:
        raise TypeError(
            f"Expected dtype float or integer. Received instead: {da.dtype}"
        )


def mask_non_spatial_var(var):
    if da.values[()] is not None:
        if is_scalar(da.values[()]):
            return da.values[()]  # For scalars, such as options
        else:  # For example for arrays with only a layer dimension
            return da
    else:
        return None

```

Reply:
I like this refactoring!

Reply:
done